Sources/Plasma/Apps/plClient/CMakeLists.txt

```diff
@@ -90,17 +89,106 @@ if(WIN32)
+	../../FeatureLib/pfMetalPipeline/ShaderSrc/WaveSet7.metal
+	../../FeatureLib/pfMetalPipeline/ShaderSrc/Grass.metal
+	../../FeatureLib/pfMetalPipeline/ShaderSrc/WaveDecEnv.metal
+    ../../FeatureLib/pfMetalPipeline/ShaderSrc/Avatar.metal
+    ../../FeatureLib/pfMetalPipeline/ShaderSrc/WaveDec1Lay_7.metal
+    ../../FeatureLib/pfMetalPipeline/ShaderSrc/WaveRip.metal
+    ../../FeatureLib/pfMetalPipeline/ShaderSrc/Clear.metal
+    ../../FeatureLib/pfMetalPipeline/ShaderSrc/GammaCorrection.metal
+    ../../FeatureLib/pfMetalPipeline/ShaderSrc/TextFontShader.metal
+    )
+    set_source_files_properties(${plClient_SHADERS} PROPERTIES LANGUAGE METAL)
+    source_group("Metal Shaders" FILES ${plClient_SHADERS})
+    set(plClient_SOURCES ${plClient_SOURCES} ${plClient_SHADERS})
+    include_directories("../../FeatureLib/pfMetalPipeline/metal-cpp"
+                        "../../FeatureLib/pfMetalPipeline/ShaderSrc")
+    list(APPEND plClient_RESOURCES
+        win32/res/resource.h
+        win32/res/banner.bmp
+        win32/res/Dirt.ICO
+        win32/res/headspin.ico
+    )
 else()
     list(APPEND plClient_SOURCES
         main.cpp
     )
 endif()
 
-plasma_executable(plClient WIN32 CLIENT INSTALL_PDB
-    SOURCES
-        ${plClient_SOURCES} ${plClient_HEADERS}
-        ${plClient_TEXT} ${plClient_RESOURCES}
-)
+if(APPLE)
+    plasma_executable(plClient CLIENT INSTALL_PDB SOURCES ${plClient_SOURCES} ${plClient_HEADERS} ${RESOURCES})
+    set(MAIN_NIBFILE "MainMenu")
+    set_target_properties(plClient PROPERTIES
+        BUNDLE True
+        MACOSX_BUNDLE_INFO_PLIST "${CMAKE_CURRENT_SOURCE_DIR}/OSX/Info.plist.in"
+        MACOSX_BUNDLE_GUI_IDENTIFIER org.openuru.plClient
+        MACOSX_BUNDLE_BUNDLE_VERSION "0.1"
+        MACOSX_BUNDLE_SHORT_VERSION_STRING "0.1"
+        XCODE_ATTRIBUTE_ASSETCATALOG_COMPILER_APPICON_NAME AppIcon;
+        XCODE_ATTRIBUTE_MTL_FAST_MATH "YES";
+        XCODE_ATTRIBUTE_MTL_ENABLE_DEBUG_INFO[variant=Debug] "INCLUDE_SOURCE";
+        XCODE_ATTRIBUTE_MTL_ENABLE_DEBUG_INFO[variant=RelWithDebInfo] "INCLUDE_SOURCE";
+        XCODE_ATTRIBUTE_GCC_GENERATE_DEBUGGING_SYMBOLS[variant=Debug] "YES";
+        XCODE_ATTRIBUTE_GCC_GENERATE_DEBUGGING_SYMBOLS[variant=RelWithDebInfo] "YES";
+    )
+    set_property (TARGET plClient APPEND_STRING PROPERTY COMPILE_FLAGS "-fobjc-arc")
+    target_sources(plClient PRIVATE OSX/Assets.xcassets)
+    set_source_files_properties(OSX/Assets.xcassets ${RESOURCES} PROPERTIES
+         MACOSX_PACKAGE_LOCATION Resources
+    )
+    target_link_libraries(plClient PRIVATE "-framework Cocoa")
+    target_link_libraries(plClient PRIVATE "-framework QuartzCore")
+    target_link_libraries(plClient PRIVATE "-framework MetalPerformanceShaders")
+    install(
+        TARGETS plClient
+        DESTINATION client
+    )
+    set (searchDirs
+        "${CMAKE_PREFIX_PATH}/lib"
+        "${CMAKE_PREFIX_PATH}/usr/local/lib"
+        "${CMAKE_PREFIX_PATH}/mac64/physx/bin/mac64.xcode/release"
+    )
```

Comment:
In theory, CMake should find and give us all the library paths we need, right?

Reply:
Yep. Seems to be working without the manual scoping.